build-logic/src/main/kotlin/DokkaPlugin.kt

```diff
@@ -0,0 +1,18 @@
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+import org.gradle.kotlin.dsl.configure
+import org.jetbrains.dokka.gradle.DokkaExtension
+
+abstract class DokkaPlugin : Plugin<Project> {
+    override fun apply(target: Project): Unit =
+        with(target) {
+            pluginManager.apply("java-library")
+            pluginManager.apply("org.jetbrains.dokka-javadoc")
+
+            extensions.configure<DokkaExtension> {
```

Comment:
The migration was pretty straightforward. I copied existing plugin and I needed to:
1. Apply all plugins by their `id`. Generated accessors like [java-library](https://github.com/pinterest/ktlint/blob/2a3c922ba6b9b34092f7782e09e779530878b8b3/build-logic/src/main/kotlin/ktlint-dokka.gradle.kts#L2) are not accessible.
2. Generated accessors for extensions don't exist anymore, so the calls to extensions like [dokka](https://github.com/pinterest/ktlint/blob/2a3c922ba6b9b34092f7782e09e779530878b8b3/build-logic/src/main/kotlin/ktlint-dokka.gradle.kts#L7) needed to be replaced with direct gradle api call
3. Kotlin DSL supported property assigments, so they all had to be replaced with plain `.set()` calls